stdlib/cgi/0/core.rbs

```diff
@@ -0,0 +1,595 @@
+  #         CGI::InvalidEncoding will be raised.
+  #
+  #         Example. Suppose `@@accept_charset` is "UTF-8"
+  #
+  #         when not specified:
+  #
+  #             cgi=CGI.new      # @accept_charset # => "UTF-8"
+  #
+  #         when specified as "EUC-JP":
+  #
+  #             cgi=CGI.new(:accept_charset => "EUC-JP") # => "EUC-JP"
+  #
+  #     `:tag_maker`
+  # :       String that specifies which version of the HTML generation methods to
+  #         use.  If not specified, no HTML generation methods will be loaded.
+  #
+  #         The following values are supported:
+  #
+  #         "html3"
+  # :           HTML 3.x
+  #         "html4"
+  # :           HTML 4.0
+  #         "html4Tr"
+  # :           HTML 4.0 Transitional
+  #         "html4Fr"
+  # :           HTML 4.0 with Framesets
+  #         "html5"
+  # :           HTML 5
+  #
+  #
+  #     `:max_multipart_length`
+  # :       Specifies maximum length of multipart data. Can be an Integer scalar
+  #         or a lambda, that will be evaluated when the request is parsed. This
+  #         allows more complex logic to be set when determining whether to accept
+  #         multipart data (e.g. consult a registered users upload allowance)
+  #
+  #         Default is 128 * 1024 * 1024 bytes
+  #
+  #             cgi=CGI.new(:max_multipart_length => 268435456) # simple scalar
+  #
+  #             cgi=CGI.new(:max_multipart_length => -> {check_filesystem}) # lambda
+  #
+  #
+  # `block`
+  # :   If provided, the block is called when an invalid encoding is encountered.
+  #     For example:
+  #
+  #         encoding_errors={}
+  #         cgi=CGI.new(:accept_charset=>"EUC-JP") do |name,value|
+  #           encoding_errors[name] = value
+  #         end
+  #
+  #
+  # Finally, if the CGI object is not created in a standard CGI call environment
+  # (that is, it can't locate REQUEST_METHOD in its environment), then it will run
+  # in "offline" mode.  In this mode, it reads its parameters from the command
+  # line or (failing that) from standard input.  Otherwise, cookies and other
+  # parameters are parsed automatically from the standard CGI locations, which
+  # varies according to the REQUEST_METHOD.
+  #
+  def initialize: [T] (?String options) ?{ (String name, String value) -> T } -> CGI
+                  | [T, U] (?Hash[Symbol, T]) ?{ (String name, String value) -> U } -> CGI
+
+  attr_reader accept_charset: String
+
+  # Return the accept character set for all new CGI instances.
+  #
+  def self.accept_charset: () -> String
+
+  # Set the accept character set for all new CGI instances.
+  #
+  def self.accept_charset=: (String accept_charset) -> String
+
+  # Parse an HTTP query string into a hash of key=>value pairs.
+  #
+  #     params = CGI.parse("query_string")
+  #       # {"name1" => ["value1", "value2", ...],
+  #       #  "name2" => ["value1", "value2", ...], ... }
+  #
+  def self.parse: [T] (String query) -> Hash[String, T]
```

Comment:
Looks like no need to use a generic method.

```suggestion
  def self.parse: (String query) -> Hash[String, String | Array[String]]   # Give precise type.
  def self.parse: (String query) -> Hash[String, untyped]                  # This would be enough.
```